From 10d4d255fc4e4bdeeca0afaea0b3405be5923508 Mon Sep 17 00:00:00 2001
From: Catalin Horghidan <catalin.horghidan@nxp.com>
Date: Thu, 22 Sep 2016 17:28:36 +0300
Subject: [PATCH 385/508] staging: dpaa2-evb: update dpdmux command ids set
 for MC v10.x

	Update dpdmux command ids in order to work with MC 10.0.0.
	The binary interface compatibility is maintained with MC v9.x.

Signed-off-by: Catalin Horghidan <catalin.horghidan@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/fsl-dpaa2/evb/dpdmux-cmd.h |   58 +++++++++++++++-------------
 1 files changed, 31 insertions(+), 27 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/evb/dpdmux-cmd.h b/drivers/staging/fsl-dpaa2/evb/dpdmux-cmd.h
index 0a5cf17..2f1c1fa 100644
--- a/drivers/staging/fsl-dpaa2/evb/dpdmux-cmd.h
+++ b/drivers/staging/fsl-dpaa2/evb/dpdmux-cmd.h
@@ -1,4 +1,4 @@
-/* Copyright 2013-2015 Freescale Semiconductor Inc.
+/* Copyright 2013-2016 Freescale Semiconductor Inc.
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -36,39 +36,43 @@
 #define DPDMUX_VER_MAJOR				5
 #define DPDMUX_VER_MINOR				0
 
+#define DPDMUX_CMD_BASE_VER				0
+#define DPDMUX_CMD_ID_OFF				4
+#define DPDMUX_CMD_ID(id) (((id) << DPDMUX_CMD_ID_OFF) | DPDMUX_CMD_BASE_VER)
+
 /* Command IDs */
-#define DPDMUX_CMDID_CLOSE				0x800
-#define DPDMUX_CMDID_OPEN				0x806
-#define DPDMUX_CMDID_CREATE				0x906
-#define DPDMUX_CMDID_DESTROY				0x900
-
-#define DPDMUX_CMDID_ENABLE				0x002
-#define DPDMUX_CMDID_DISABLE				0x003
-#define DPDMUX_CMDID_GET_ATTR				0x004
-#define DPDMUX_CMDID_RESET				0x005
-#define DPDMUX_CMDID_IS_ENABLED				0x006
-
-#define DPDMUX_CMDID_SET_IRQ				0x010
-#define DPDMUX_CMDID_GET_IRQ				0x011
-#define DPDMUX_CMDID_SET_IRQ_ENABLE			0x012
-#define DPDMUX_CMDID_GET_IRQ_ENABLE			0x013
-#define DPDMUX_CMDID_SET_IRQ_MASK			0x014
-#define DPDMUX_CMDID_GET_IRQ_MASK			0x015
-#define DPDMUX_CMDID_GET_IRQ_STATUS			0x016
-#define DPDMUX_CMDID_CLEAR_IRQ_STATUS			0x017
-
-#define DPDMUX_CMDID_UL_SET_MAX_FRAME_LENGTH		0x0a1
-
-#define DPDMUX_CMDID_UL_RESET_COUNTERS			0x0a3
-
-#define DPDMUX_CMDID_IF_SET_ACCEPTED_FRAMES		0x0a7
-#define DPDMUX_CMDID_IF_GET_ATTR			0x0a8
-
-#define DPDMUX_CMDID_IF_ADD_L2_RULE			0x0b0
-#define DPDMUX_CMDID_IF_REMOVE_L2_RULE			0x0b1
-#define DPDMUX_CMDID_IF_GET_COUNTER			0x0b2
-#define DPDMUX_CMDID_IF_SET_LINK_CFG		0x0b3
-#define DPDMUX_CMDID_IF_GET_LINK_STATE		0x0b4
+#define DPDMUX_CMDID_CLOSE                   DPDMUX_CMD_ID(0x800)
+#define DPDMUX_CMDID_OPEN                    DPDMUX_CMD_ID(0x806)
+#define DPDMUX_CMDID_CREATE                  DPDMUX_CMD_ID(0x906)
+#define DPDMUX_CMDID_DESTROY                 DPDMUX_CMD_ID(0x900)
+
+#define DPDMUX_CMDID_ENABLE                  DPDMUX_CMD_ID(0x002)
+#define DPDMUX_CMDID_DISABLE                 DPDMUX_CMD_ID(0x003)
+#define DPDMUX_CMDID_GET_ATTR                DPDMUX_CMD_ID(0x004)
+#define DPDMUX_CMDID_RESET                   DPDMUX_CMD_ID(0x005)
+#define DPDMUX_CMDID_IS_ENABLED              DPDMUX_CMD_ID(0x006)
+
+#define DPDMUX_CMDID_SET_IRQ                 DPDMUX_CMD_ID(0x010)
+#define DPDMUX_CMDID_GET_IRQ                 DPDMUX_CMD_ID(0x011)
+#define DPDMUX_CMDID_SET_IRQ_ENABLE          DPDMUX_CMD_ID(0x012)
+#define DPDMUX_CMDID_GET_IRQ_ENABLE          DPDMUX_CMD_ID(0x013)
+#define DPDMUX_CMDID_SET_IRQ_MASK            DPDMUX_CMD_ID(0x014)
+#define DPDMUX_CMDID_GET_IRQ_MASK            DPDMUX_CMD_ID(0x015)
+#define DPDMUX_CMDID_GET_IRQ_STATUS          DPDMUX_CMD_ID(0x016)
+#define DPDMUX_CMDID_CLEAR_IRQ_STATUS        DPDMUX_CMD_ID(0x017)
+
+#define DPDMUX_CMDID_UL_SET_MAX_FRAME_LENGTH DPDMUX_CMD_ID(0x0a1)
+
+#define DPDMUX_CMDID_UL_RESET_COUNTERS       DPDMUX_CMD_ID(0x0a3)
+
+#define DPDMUX_CMDID_IF_SET_ACCEPTED_FRAMES  DPDMUX_CMD_ID(0x0a7)
+#define DPDMUX_CMDID_IF_GET_ATTR             DPDMUX_CMD_ID(0x0a8)
+
+#define DPDMUX_CMDID_IF_ADD_L2_RULE          DPDMUX_CMD_ID(0x0b0)
+#define DPDMUX_CMDID_IF_REMOVE_L2_RULE       DPDMUX_CMD_ID(0x0b1)
+#define DPDMUX_CMDID_IF_GET_COUNTER          DPDMUX_CMD_ID(0x0b2)
+#define DPDMUX_CMDID_IF_SET_LINK_CFG         DPDMUX_CMD_ID(0x0b3)
+#define DPDMUX_CMDID_IF_GET_LINK_STATE       DPDMUX_CMD_ID(0x0b4)
 
 /*                cmd, param, offset, width, type, arg_name */
 #define DPDMUX_CMD_OPEN(cmd, dpdmux_id) \
-- 
1.7.5.4

